backends/fs/atomic.go

```diff
@@ -0,0 +1,90 @@
+package fs
```

Comment:
This looks solid, I guess you have looked at other similar solutions like this one:
https://github.com/google/renameio

just to see we are not missing any corner cases?

Reply:
Thanks for the recommendation. I gave it a look and while it looks like it can handle a broader set of situations (that we don't need here), I didn't see something that would benefit us here.